### PR TITLE
OCTOPUS-625:Moved Ignition and templates files to virtual machine folder

### PR DESCRIPTION
**Pull request by pkenchap**
Removed the ignition roles and added as part of virtual machine task . 
Modified the templates for virtual machine .

### CODE REVIEW

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml [hunk 1]
```diff
@@ -21,3 +23,18 @@ virtual_machine_create_userdata: "#!/bin/sh\nyum -y install python3"
 #           - ansible
 #         package_upgrade: true
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml:24]:
```suggestion
#         package_upgrade: true
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
-virtual_machine_create_flavor: "PUN_custom_flavor"
+virtual_machine_create_image_name: 4df699b6-6877-4b77-9ddb-0d6dc9cefb68
+virtual_machine_create_flavor: "m1.medium"
 virtual_machine_create_volume_size: 80
 virtual_machine_create_keypair_name: pun_keypair
 virtual_machine_create_network_name: provider
-virtual_machine_create_userdata: "#!/bin/sh\nyum -y install python3"
+virtual_machine_create_userdata: "{{ lookup('template', 'worker-amd64.ign.j2') }}"
+# virtual_machine_create_userdata: "{{ lookup('file', '/var/www/html/ignition/worker-amd64.ign') | string }}"
+# virtual_machine_create_userdata: "#!/bin/sh\nyum -y install python3"
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml:13]:
```suggestion
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml [hunk 3]
```diff
@@ -3,12 +3,14 @@
...
-virtual_machine_create_image_name: 0c4d9ba3-ed14-403a-a35d-8b4c9f720f1f
-virtual_machine_create_flavor: "PUN_custom_flavor"
+virtual_machine_create_image_name: 4df699b6-6877-4b77-9ddb-0d6dc9cefb68
+virtual_machine_create_flavor: "m1.medium"
 virtual_machine_create_volume_size: 80
 virtual_machine_create_keypair_name: pun_keypair
 virtual_machine_create_network_name: provider
-virtual_machine_create_userdata: "#!/bin/sh\nyum -y install python3"
+virtual_machine_create_userdata: "{{ lookup('template', 'worker-amd64.ign.j2') }}"
+# virtual_machine_create_userdata: "{{ lookup('file', '/var/www/html/ignition/worker-amd64.ign') | string }}"
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml:12]:
```suggestion
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml (hunk 1 above)
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml:24]:
```suggestion
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml [hunk 4]
```diff
@@ -21,3 +23,18 @@ virtual_machine_create_userdata: "#!/bin/sh\nyum -y install python3"
 #           - ansible
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/defaults/main.yml:23]:
```suggestion
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ---
 - name: Create worker.ign and store it to /var/www/html/ignition/
   ansible.builtin.get_url:
-    url: https://{{ bastion_ip }}:22623/config/worker
+    url: https://{{ virtual_machine_create_bastion_ip }}:22623/config/worker
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:4]:
```suggestion
    url: https://{{ bastion_ip }}:22623/config/worker
```

Comment (prb112) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:4]: let's just call it bastion_ip

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml [hunk 6]
```diff
@@ -14,8 +14,8 @@
 
 - name: Set the variables (To be used in Customization Script)
   ansible.builtin.set_fact:
-    worker_hostname_encoded: "{{ worker_hostname | b64encode }}"
-    dns_none_encoded: "{{ dns_none_conf | b64encode }}"
+    worker_hostname_encoded: "{{ virtual_machine_create_worker_hostname | b64encode }}"
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:17]:
```suggestion
    worker_hostname_encoded: "{{ worker_hostname | b64encode }}"
```

Comment (prb112) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:17]: please use worker_hostname. it's more readable

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml [hunk 7]
```diff
@@ -14,8 +14,8 @@
...
+    dns_none_encoded: "{{ virtual_machine_create_dns_none_conf | b64encode }}"
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:18]: please use dns_none_conf

Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:18]:
```suggestion
    dns_none_encoded: "{{ dns_none_conf | b64encode }}"
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/main.yml [hunk 8]
```diff
@@ -1,5 +1,8 @@
 ---
 # file: intel-worker-playbook.yml
+- name: Add identity resources
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/main.yml:3]:
```suggestion
- name: Add ignition resources
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/resolv.conf.j2 [hunk 9]
```diff
@@ -0,0 +1,2 @@
+search {{ virtual_machine_create_domain_name }}
+nameserver {{ virtual_machine_create_bastion_ip }}
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/resolv.conf.j2:2]:
```suggestion
search {{ domain_name }}
nameserver {{ bastion_ip }}
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2 [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+					"verification": {}
+				},
+				"mode": 420
+		},
+		{
+				"group": {},
+				"path": "/etc/NetworkManager/conf.d/90-dns-none.conf",
+				"user": {},
+				"contents": {
+					"source": "data:text/plain;base64,{{ virtual_machine_create_dns_none_encoded }}",
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2:42]:
```suggestion
					"source": "data:text/plain;base64,{{ dns_none_encoded }}",
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2 [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+					"verification": {}
+				},
+				"mode": 420
+		},
+		{
+				"group": {},
+				"path": "/etc/resolv.conf",
+				"user": {},
+				"contents": {
+					"source": "data:text/plain;base64,{{ virtual_machine_create_etc_resolve_encoded }}",
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2:32]:
```suggestion
					"source": "data:text/plain;base64,{{ etc_resolve_encoded }}",
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2 [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+		"httpTotal": 15
+	},
+	"storage": {
+	  "files": [
+		{
+				"group": {},
+				"path": "/etc/hostname",
+				"user": {},
+				"contents": {
+					"source": "data:text/plain;base64,{{ virtual_machine_create_worker_hostname_encoded }}",
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2:22]:
```suggestion
					"source": "data:text/plain;base64,{{ hostname_encoded }}",
```

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2 [hunk 13]
```diff
@@ -0,0 +1,49 @@
+{
+	"ignition": {
+		"version": "3.4.0",
+		"config": {
+			"merge": [
+				{
+					"source": "http://{{ virtual_machine_create_bastion_ip }}:{{ virtual_machine_create_http_port }}/ignition/worker-amd64.ign"
```
Comment (prb112) [inline openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2:7]:
```suggestion
					"source": "http://{{ bastion_ip }}:{{ http_port }}/ignition/worker-amd64.ign"
```

**Comment by prb112** [comment]
Hey Punith, 

Made some comments based on readability. 
Please address when you can.

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml (hunk 5 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:4]: Ansible lint expects to have a role name as a prefix for the variable names . if we just keep bastion_ip it was giving error .

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml (hunk 6 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:17]: Ansible lint expects to have a role name as a prefix for the variable names .

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml (hunk 7 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:18]: same as above .

Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/ignition.yml:18]: Ansible lint expects to have a role name as a prefix for the variable names .

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/resolv.conf.j2 (hunk 9 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/resolv.conf.j2:2]: Ansible lint expects to have a role name as a prefix for the variable names .

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/main.yml (hunk 8 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/tasks/main.yml:3]: Sure , I will make these changes.

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2 (hunk 13 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2:7]: Ansible lint expects to have a role name as a prefix for the variable names .

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2 (hunk 12 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2:22]: Ansible lint expects to have a role name as a prefix for the variable names .

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2 (hunk 11 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2:32]: Ansible lint expects to have a role name as a prefix for the variable names .

File: openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2 (hunk 10 above)
Comment (pkenchap) [reply openstack/intel-worker/playbooks/roles/virtual_machine_create/templates/worker-amd64.ign.j2:42]: Ansible lint expects to have a role name as a prefix for the variable names .

**Comment by prb112** [comment]
Discussed with Punith the restrictions on variable names
/LGTM

**Pull request merged**